File: scripts/python/lib/commands/comment.py
from ..args import parse_args, require_positional, to_int
from ..errors import AppError
from ..constants import ERROR_CODES
from ..utils import read_text_file_from_cwd

NAME = 'comment'

def run_add(args, context):
    parsed = parse_args(args)
    issue_key = require_positional(parsed['positionals'], 0, 'ISSUE-KEY')
    
    body = None
    if parsed['options'].get('body-file'):
        body = read_text_file_from_cwd(parsed['options']['body-file'])
    else:
        body = parsed['options'].get('body')

    if not body:
        raise AppError(ERROR_CODES.VALIDATION, 'comment add requires --body or --body-file.')

    if context['global_options'].get('dry_run'):
        return {
            'data': {
                'ok': True,
                'dryRun': True,
                'action': 'comment.add',
                'issueKey': issue_key,
                'request': {'body': body}
            }
        }

    data = context['client'].add_comment(issue_key, body)
    return {
        'data': data,
        'tableRows': [
            {
                'issueKey': issue_key,
                'id': data.get('id'),
                'author': data.get('author', {}).get('displayName') if data.get('author') else '',
                'created': data.get('created')
            }
        ]
    }

def run_list(args, context):
    parsed = parse_args(args)
    issue_key = require_positional(parsed['positionals'], 0, 'ISSUE-KEY')
    start_at = to_int(parsed['options'].get('start-at'), 0, '--start-at')
    max_results = to_int(parsed['options'].get('max-results'), 50, '--max-results')
    
    data = context['client'].list_comments(issue_key, start_at=start_at, max_results=max_results)
    comments = data.get('comments') or []
    
    return {
        'data': data,
        'tableRows': [
            {
                'id': item.get('id'),
                'author': item.get('author', {}).get('displayName') if item.get('author') else '',
                'created': item.get('created'),
                'body': str(item.get('body') or '').replace('\n', ' ').replace('\r', ' ')[:80]
            } for item in comments
        ]
    }

SUBCOMMANDS = {
    'add': run_add,
    'list': run_list
}

def run(args, context):
    if not args:
        raise AppError(ERROR_CODES.VALIDATION, 'comment requires subcommand add|list.')
    
    subcommand = args[0]
    rest = args[1:]
    
    if subcommand not in SUBCOMMANDS:
        raise AppError(ERROR_CODES.VALIDATION, 'comment requires subcommand add|list.')
    
    return SUBCOMMANDS[subcommand](rest, context)

File: scripts/lib/commands/comment.js
const { parseArgs, requirePositional, toInt } = require('../args');
const { AppError } = require('../errors');
const { ERROR_CODES } = require('../constants');
const { readTextFileFromCwd } = require('../utils');

async function runAdd(args, context) {
  const parsed = parseArgs(args, {});
  const issueKey = requirePositional(parsed.positionals, 0, 'ISSUE-KEY');
  const body = parsed.options['body-file']
    ? readTextFileFromCwd(parsed.options['body-file'])
    : parsed.options.body;

  if (!body) {
    throw new AppError(ERROR_CODES.VALIDATION, 'comment add requires --body or --body-file.');
  }

  if (context.globalOptions.dryRun) {
    return {
      data: {
        ok: true,
        dryRun: true,
        action: 'comment.add',
        issueKey,
        request: { body }
      }
    };
  }

  const data = await context.client.addComment(issueKey, body);
  return {
    data,
    tableRows: [
      {
        issueKey,
        id: data.id,
        author: data.author ? data.author.displayName : '',
        created: data.created
      }
    ]
  };
}

async function runList(args, context) {
  const parsed = parseArgs(args, {});
  const issueKey = requirePositional(parsed.positionals, 0, 'ISSUE-KEY');
  const startAt = toInt(parsed.options['start-at'], 0, '--start-at');
  const maxResults = toInt(parsed.options['max-results'], 50, '--max-results');
  const data = await context.client.listComments(issueKey, startAt, maxResults);
  const comments = data.comments || [];
  return {
    data,
    tableRows: comments.map((item) => ({
      id: item.id,
      author: item.author ? item.author.displayName : '',
      created: item.created,
      body: String(item.body || '').replace(/\s+/g, ' ').slice(0, 80)
    }))
  };
}

const subcommands = new Map([
  ['add', runAdd],
  ['list', runList]
]);

async function run(args, context) {
  const [subcommand, ...rest] = args;
  if (!subcommand || !subcommands.has(subcommand)) {
    throw new AppError(ERROR_CODES.VALIDATION, 'comment requires subcommand add|list.');
  }
  return subcommands.get(subcommand)(rest, context);
}

module.exports = {
  name: 'comment',
  description: 'Comment operations (add/list)',
  run
};
